ginga/canvas/types/astro.py

```diff
@@ -666,7 +679,238 @@ def move_to(self, xdst, ydst):
+
+    def _calc_axes(self, viewer, image, rot_deg, swapxy):
+        self._cur_image = image
+        self._cur_rot = rot_deg
+        self._cur_swap = swapxy
+
+        if not isinstance(image, AstroImage) or not image.has_valid_wcs():
+            return []
+
+        # Approximate bounding box in RA/DEC space
+        xmax = image.width - 1
+        ymax = image.height - 1
+        try:
+            radec = image.wcs.pointstocoords(
+                [[0, 0], [0, ymax], [xmax, 0], [xmax, ymax]],
+                naxispath=image.naxispath)
+        except Exception:
+            return []
+        ra_min, dec_min = radec.ra.min().deg, radec.dec.min().deg
+        ra_max, dec_max = radec.ra.max().deg, radec.dec.max().deg
+        ra_size = ra_max - ra_min
+        dec_size = dec_max - dec_min
+
+        # Calculate positions of RA/DEC lines
+        d_ra = ra_size / (self.num_ra + 1)
+        d_dec = dec_size / (self.num_dec + 1)
+        ra_arr = numpy.arange(ra_min + d_ra, ra_max - d_ra * 0.5, d_ra)
+        dec_arr = numpy.arange(dec_min + d_dec, dec_max - d_ra * 0.5, d_dec)
+
+        # RA/DEC step size for each vector
+        min_imsize = min(image.width, image.height)
+        d_ra_step = ra_size * self._pix_res / min_imsize
+        d_dec_step = dec_size * self._pix_res / min_imsize
+
+        # Create Path objects
+        objs = []
+
+        for cur_ra in ra_arr:
+            crds = [[cur_ra, cur_dec] for cur_dec in
+                    numpy.arange(dec_min, dec_max + d_dec_step, d_dec_step)]
+            lbl = raDegToString(cur_ra)
+            objs += self._get_path(viewer, image, crds, lbl, 1)
+        for cur_dec in dec_arr:
+            crds = [[cur_ra, cur_dec] for cur_ra in
+                    numpy.arange(ra_min, ra_max + d_ra_step, d_ra_step)]
+            lbl = decDegToString(cur_dec)
+            objs += self._get_path(viewer, image, crds, lbl, 0)
+
+        return objs
+
+    def _get_path(self, viewer, image, crds, lbl, axis):
+        from ginga.canvas.types.basic import Path, Text
+
+        try:
+            pts = image.wcs.skytopoints(crds, naxispath=image.naxispath)
+        except Exception:
+            return []
+
+        # Don't draw outside image area
+        mask = ((pts[:, 0] >= 0) & (pts[:, 0] < image.width) &
+                (pts[:, 1] >= 0) & (pts[:, 1] < image.height))
+        pts = pts[mask]
+
+        path_obj = Path(
+            points=pts, coords='data', linewidth=self.linewidth,
+            linestyle=self.linestyle, color=self.color,
+            alpha=self.alpha)
+        # this is necessary because we are not actually adding to a canvas
+        path_obj.crdmap = viewer.get_coordmap('data')
+
+        if self.show_label:
+            # Calculate label orientation
+            x1, y1 = pts[0]
+            x2, y2 = pts[-1]
+            dx = x2 - x1
+            dy = y2 - y1
+            m = dy / dx
+            c = y1 - m * x1
+
+            if abs(m) < 1:  # x axis varying
```

Comment:
Note: RA and DEC don't always correspond with X/Y. Sometimes the image sky orientation is at an angle w.r.t. cartesian coordinates.